src/Board.cpp

```diff
@@ -0,0 +1,211 @@
+    */
+    // White pieces
+    m_board[0][0] = new Rook(Piece::Color::WHITE, char(Row::START_ROW_WHITE), int(Column::COLUMN_1));
+    //m_board[0][1] = new Knight(Piece::Color::WHITE, char(Row::PAWN_START_ROW_WHITE), int(Column::COLUMN_2));
+    m_board[0][2] = new Bishop(Piece::Color::WHITE, char(Row::START_ROW_WHITE), int(Column::COLUMN_3));
+    m_board[0][3] = new Queen(Piece::Color::WHITE, char(Row::START_ROW_WHITE), int(Column::COLUMN_4));
+    m_board[0][4] = new King(Piece::Color::WHITE, char(Row::START_ROW_WHITE), int(Column::COLUMN_5));
+    m_board[0][5] = new Bishop(Piece::Color::WHITE, char(Row::START_ROW_WHITE), int(Column::COLUMN_6));
+    //m_board[0][6] = new Knight(Piece::Color::WHITE, char(Row::START_ROW_WHITE), int(Column::COLUMN_7));
+    m_board[0][7] = new Rook(Piece::Color::WHITE, char(Row::START_ROW_WHITE), int(Column::COLUMN_8));
+    /*
+    for (int i = 0; i < BOARD_SIZE; i++) {
+        m_board[1][i] = new Pawn(Piece::Color::WHITE, char(Row::PAWN_START_ROW_WHITE), i + 1);
+    }
+    */
+
+    // fill in m_black_pieces and m_white_pieces.
+    for (int i = 0; i < BOARD_SIZE; i++) {
+        for (int j = 0; j < BOARD_SIZE; j++) {
+            if (m_board[i][j] != nullptr) {
+                if (m_board[i][j]->getColor() == Piece::Color::BLACK) {
+                    m_black_pieces.push_back(m_board[i][j]);
+                }
+                else {
+                    m_white_pieces.push_back(m_board[i][j]);
+                }
+            }
+        }
+    }
+
+}
+
+// Return a vector of pointers to all the pieces on the board with the given color
+std::vector<Piece*> Board::getPieces(Piece::Color color) const {
+    std::vector<Piece*> pieces;
+    for (const auto& row : m_board) {
+        for (const auto& piece : row) {
+            if (piece && piece->getColor() == color) {
+                pieces.push_back(piece);
+            }
+        }
+    }
+    return pieces;
+}
+
+void Board::movePiece(const char from_row, const int from_col, const char to_row, const int to_col) {
+    // Update the piece's position
+    Piece* piece = m_board[from_row - 'a'][from_col - 1];
+    piece->updatePos(to_row, to_col);
+
+    // Update the board state
+    m_board[from_row - 'a'][from_col - 1] = nullptr;
+    m_board[to_row - 'a'][to_col - 1] = piece;
+}
+
+std::vector<std::vector<Piece*>> Board::getPieces() const {
+    return m_board;
+}
+
+bool Board::isKingInCheck(Piece::Color color, char from_row, int from_col, char to_row, int to_col) const {
+    // Create a copy of m_board called board_after_move
+    auto board_after_move = m_board;
+
+    // Move the piece from [from_row][from_col] to [to_row][to_col] in board_after_move
+    board_after_move[to_row - 'a'][to_col - 1] = board_after_move[from_row - 'a'][from_col - 1];
+    board_after_move[from_row - 'a'][from_col - 1] = nullptr;
+
+    // Find the position of the king of the given color
+    char kingRow;
+    int kingCol;
+    for (const auto& row : m_board) {
```

Comment:
This is a good example of how you can use functions to break up your code to small logic pieces. 
So in this case, instead of putting all this logic here you could create a separate function called something like `GetKingLocation(Color..)` and then in this function just call that function.
Writing like this makes your code more readable and easier to maintain